From 7e8541b08d10af5a90a1e021f60f8e5acf2f4679 Mon Sep 17 00:00:00 2001
From: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
Date: Wed, 19 Jul 2017 10:38:42 +0530
Subject: [PATCH 019/155] drivers:net:fsl-mc: Update MC address calculation

Update MC address caluclation as per MC design requirement
of address as least significant 512MB address
of MC private allocated memory.

Signed-off-by: Priyanka Jain <priyanka.jain@nxp.com>
Signed-off-by: Ashish Kumar <ashish.kumar@nxp.com>
---
 drivers/net/fsl-mc/mc.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/fsl-mc/mc.c b/drivers/net/fsl-mc/mc.c
index 8bf25c7040..977c17d321 100644
--- a/drivers/net/fsl-mc/mc.c
+++ b/drivers/net/fsl-mc/mc.c
@@ -802,10 +802,15 @@ int get_dpl_apply_status(void)
 
 /**
  * Return the MC address of private DRAM block.
+ * MC address should be least significant 512MB address
+ * of MC private memory
  */
 u64 mc_get_dram_addr(void)
 {
-	return gd->arch.resv_ram;
+	size_t mc_ram_size = mc_get_dram_block_size();
+
+	return (gd->arch.resv_ram + mc_ram_size - 1) &
+		MC_RAM_BASE_ADDR_ALIGNMENT_MASK;
 }
 
 /**
-- 
2.14.1

